qiskit/pulse/library/discrete.py

```diff
@@ -65,6 +83,18 @@ def zero(duration: int, name: Optional[str] = None) -> Waveform:
 _sampled_square_pulse = samplers.midpoint(continuous.square)
 
 
+@deprecate_func(
+    since="0.25.0",
+    additional_msg="The discrete pulses library, including square() is pending deprecation."
+    " Instead, use the SymbolicPulse library to create the waveform with"
+    " pulse.Square(...).get_waveform()."
+    " Note that pulse.Square() does not support complex values for `amp`."
+    " Instead, use two float values for (`amp`, `angle`)."
+    " Also note that the underlying `sign` function used to"
```

Comment:
I cannot understand this message without reading code. Maybe we can say new symbolic implementation may return different value at t=0?

Reply:
Then you can write more details (possibly with example code) in release note and say "read release note for more details" here? I don't really like lengthy warning as it might be raised many times.

Reply:
I opted for matching the functional form of `square()` and `Square()`, so this discussion was almost mute. However, when I did that I noticed that the phase in `square()` was defined somewhat strangely, so I changed that, and now the warning addresses that - but it points the user to the documentation like you suggested.